exporters/FusionRobotExporter/Source/Data/BXDJ/ConfigData.cpp

```diff
@@ -178,164 +178,126 @@ void ConfigData::filterJoints(std::vector<core::Ptr<fusion::AsBuiltJoint>> filte
 		joints.erase(jointToErase);
 }
 
-rapidjson::Value ConfigData::getJSONObject(rapidjson::MemoryPoolAllocator<>& allocator) const
+nlohmann::json ConfigData::getJSONObject() const
 {
-	rapidjson::Value configJSON;
-	configJSON.SetObject();
-
-	// General Information
-	configJSON.AddMember("name", rapidjson::Value(robotName.c_str(), robotName.length(), allocator), allocator);
-	configJSON.AddMember("drivetrainType", rapidjson::Value((int)drivetrainType), allocator);
-	std::string convex = "BOX";
-	switch (convexType) {
-	case BOX:
-		convex = "BOX";
-		configJSON.AddMember("convex", rapidjson::Value(convex.c_str(), convex.length(), allocator), allocator);
-		break;
-	case VHACD_LOW:
-		convex = "VHACD_LOW";
-		configJSON.AddMember("convex", rapidjson::Value(convex.c_str(), convex.length(), allocator), allocator);
-		break;
-
-	case VHACD_MID:
-		convex = "VHACD_MID";
-		configJSON.AddMember("convex", rapidjson::Value(convex.c_str(), convex.length(), allocator), allocator);
-		break;
-
-	case VHACD_HIGH:
-		convex = "VHACD_HIGH";
-		configJSON.AddMember("convex", rapidjson::Value(convex.c_str(), convex.length(), allocator), allocator);
-		break;
+	nlohmann::json configJSON;
+	configJSON["name"] = robotName;
+	configJSON["driveTrainType"] = (int)drivetrainType;
+	configJSON["convex"] = (int)convexType;
+
 
-	}
 
+	configJSON["weight"] = weight;
 
-	// Weight
-	configJSON.AddMember("weight", rapidjson::Value(weight), allocator);
 
-	configJSON.AddMember("tempIconDir", rapidjson::Value(tempIconDir.c_str(), tempIconDir.length(), allocator), allocator);
+	configJSON["tempIconDir"] = tempIconDir;
 
 	// Joints
-	rapidjson::Value jointsJSON;
-	jointsJSON.SetArray();
+	nlohmann::json jointsJson = nlohmann::json::array();
+	
 
 	for (auto i = joints.begin(); i != joints.end(); i++)
 	{
 		std::string jointID = i->first;
 		JointConfig jointConfig = i->second;
 
-		rapidjson::Value jointJSON;
-		jointJSON.SetObject();
-
+		nlohmann::json jointJSON;
+		
 		// Joint Info
-		jointJSON.AddMember("id", rapidjson::Value(jointID.c_str(), jointID.length(), allocator), allocator);
-		jointJSON.AddMember("name", rapidjson::Value(jointConfig.name.c_str(), jointConfig.name.length(), allocator), allocator);
-		jointJSON.AddMember("asBuilt", jointConfig.asBuilt, allocator);
-		jointJSON.AddMember("type", rapidjson::Value((int)jointConfig.motion), allocator);
-
-
-
+		jointJSON["id"] = jointID;
+		jointJSON["name"] = jointConfig.name;
```

Comment:
Could you add the joint weight to the JSON parsing?

Reply:
Ill do that in a new branch where we do per joint weights. Not fitting in this one.